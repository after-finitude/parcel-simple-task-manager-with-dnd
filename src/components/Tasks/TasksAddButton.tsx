import React, { useCallback } from "react";
import { useDispatch } from "react-redux";
import { styled } from "@material-ui/core/styles";
import AddIcon from "@material-ui/icons/Add";
import Button from "@material-ui/core/Button";

import { openDialog } from "store/dialog/actions";
import { DialogTypes } from "store/dialog/types";

export const TasksAddButton: React.FC = () => {
  const dispatch = useDispatch();

  const handleAddButton = useCallback(
    () => dispatch(openDialog(DialogTypes.CREATE)),
    [dispatch],
  );

  return <AddButton onClick={handleAddButton}>Add task</AddButton>;
};

const AddButton = styled(props => (
  <Button startIcon={<AddIcon />} {...props} />
))({
  width: "100%",
});
