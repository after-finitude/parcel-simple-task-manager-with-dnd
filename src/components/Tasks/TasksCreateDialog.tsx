import React, { useCallback, memo } from "react";
import { useDispatch } from "react-redux";
import { useForm } from "react-hook-form";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

import { closeDialog } from "store/dialog/actions";
import { DialogTypes } from "store/dialog/types";
import { thunkAddTask } from "store/tasks/thunks";

type FormData = {
  title: string;
  description: string;
};

interface Props {
  status?: boolean;
}

export const TasksCreateDialog: React.FC<Props> = memo(({ status = false }) => {
  const dispatch = useDispatch();
  const { register, handleSubmit } = useForm<FormData>();

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const onSubmit = useCallback(
    handleSubmit(({ title, description }) => {
      dispatch(thunkAddTask(title, description));
    }),
    [dispatch, thunkAddTask],
  );
  const handleClose = useCallback(
    () => dispatch(closeDialog(DialogTypes.CREATE)),
    [dispatch],
  );

  return (
    <Dialog
      onClose={handleClose}
      open={status}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">Create task</DialogTitle>
      <form onSubmit={onSubmit}>
        <DialogContent>
          <TextField
            required
            autoFocus
            margin="dense"
            name="title"
            label="Title"
            variant="outlined"
            fullWidth
            inputRef={register}
          />
          <TextField
            margin="dense"
            name="description"
            label="Description"
            variant="outlined"
            fullWidth
            inputRef={register}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Cancel
          </Button>
          <Button type="submit" color="primary">
            Add task
          </Button>
        </DialogActions>
      </form>
    </Dialog>
  );
});
