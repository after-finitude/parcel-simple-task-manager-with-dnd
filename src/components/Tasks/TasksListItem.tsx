import React, { memo, useCallback } from "react";
import { useDispatch } from "react-redux";
import styled from "styled-components";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
import EditIcon from "@material-ui/icons/Edit";
import RestoreIcon from "@material-ui/icons/Restore";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";

import { openDialog } from "store/dialog/actions";
import { DialogTypes } from "store/dialog/types";
import { closeTask, restoreTask } from "store/tasks/actions";

interface ComponentProps {
  id: string;
  title: string;
  description?: string;
  closed: boolean;
}

export const TasksListItem: React.FC<ComponentProps> = memo(
  ({ id, title, description = "", closed }) => {
    const dispatch = useDispatch();

    const handleEdit = useCallback(
      () => dispatch(openDialog(DialogTypes.UPDATE, id)),
      [dispatch, id],
    );
    const handleClose = useCallback(() => dispatch(closeTask(id)), [
      dispatch,
      id,
    ]);
    const handleRestore = useCallback(() => dispatch(restoreTask(id)), [
      dispatch,
      id,
    ]);

    return (
      <ListItemWrapper closed={closed}>
        <ListItem>
          <ListItemText primary={title} secondary={description} />
          <ListItemSecondaryAction>
            <IconButton
              disabled={closed}
              onClick={handleEdit}
              edge="end"
              aria-label="edit"
            >
              <EditIcon />
            </IconButton>
            {closed ? (
              <IconButton
                onClick={handleRestore}
                edge="end"
                aria-label="restore"
              >
                <RestoreIcon />
              </IconButton>
            ) : (
              <IconButton onClick={handleClose} edge="end" aria-label="close">
                <CloseIcon />
              </IconButton>
            )}
          </ListItemSecondaryAction>
        </ListItem>
      </ListItemWrapper>
    );
  },
);

const ListItemWrapper = styled.div<{ closed: boolean }>`
  border: 1px grey solid;
  margin-bottom: 5px;
  opacity: ${(props: { closed: boolean }) => (props.closed ? 0.5 : 1)};
`;
