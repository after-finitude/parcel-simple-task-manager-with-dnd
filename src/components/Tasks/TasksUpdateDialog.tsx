import React, { useCallback, memo } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useForm } from "react-hook-form";

import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

import { RootState } from "store/rootReducer";
import { DialogTypes } from "store/dialog/types";
import { closeDialog } from "store/dialog/actions";
import { selectTaskById } from "store/dialog/selectors";
import { thunkUpdateTask } from "store/tasks/thunks";
import { Task } from "store/tasks/types";

type FormData = {
  title: string;
  description: string;
};

interface Props {
  status: boolean;
  currentTaskId: string;
}

export const TasksUpdateDialog: React.FC<Props> = memo(
  ({ status, currentTaskId }) => {
    const dispatch = useDispatch();
    const { register, handleSubmit } = useForm<FormData>();
    const currentTask = useSelector<RootState, Task | undefined>(
      selectTaskById(currentTaskId)
    );

    const onSubmit = useCallback(
      handleSubmit(({ title, description }) => {
        dispatch(thunkUpdateTask(title, description));
      }),
      [dispatch, thunkUpdateTask]
    );
    const handleClose = useCallback(
      () => dispatch(closeDialog(DialogTypes.UPDATE)),
      [dispatch]
    );

    return (
      <Dialog
        onClose={handleClose}
        open={status}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Update task</DialogTitle>
        <form onSubmit={onSubmit}>
          <DialogContent>
            <TextField
              defaultValue={currentTask?.title ?? ""}
              autoFocus
              margin="dense"
              name="title"
              label="Title"
              variant="outlined"
              fullWidth
              inputRef={register}
            />
            <TextField
              defaultValue={currentTask?.description ?? ""}
              margin="dense"
              name="description"
              label="Description"
              variant="outlined"
              fullWidth
              inputRef={register}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={handleClose} color="primary">
              Cancel
            </Button>
            <Button type="submit" color="primary">
              Update task
            </Button>
          </DialogActions>
        </form>
      </Dialog>
    );
  }
);
