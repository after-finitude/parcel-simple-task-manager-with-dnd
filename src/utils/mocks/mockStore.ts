import configureStore from "redux-mock-store";

const store = configureStore();

export const mockStore = store({
  tasks: [
    { id: "test1", title: "Test1", description: "test1", closed: false },
    { id: "test2", title: "Test2", description: "test2", closed: false },
  ],
  dialog: { type: "", status: false, currentTaskId: "" },
});
