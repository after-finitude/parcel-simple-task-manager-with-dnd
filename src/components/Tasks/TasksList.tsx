import React, { useState, useEffect, memo, useCallback } from "react";
import { useSelector, useDispatch, shallowEqual } from "react-redux";
import { ReactSortable } from "react-sortablejs";
import List from "@material-ui/core/List";

import { RootState } from "store/rootReducer";
import { TasksState, Task } from "store/tasks/types";
import { updateState } from "store/tasks/actions";

import { TasksListItem } from "components/Tasks/TasksListItem";

export const TasksList: React.FC = memo(() => {
  const dispatch = useDispatch();
  const storeTasks = useSelector<RootState, TasksState>(
    (state) => state.tasks,
    shallowEqual
  );
  const [tasks, setTasks] = useState<TasksState>(storeTasks);

  useEffect(() => {
    setTasks(storeTasks);
  }, [storeTasks]);

  const onEnd = useCallback(() => dispatch(updateState(tasks)), [
    dispatch,
    tasks,
  ]);

  return (
    <List>
      <ReactSortable onEnd={onEnd} list={tasks} setList={setTasks}>
        {tasks.map((task: Task) => (
          <TasksListItem
            key={task?.id}
            id={task?.id}
            title={task?.title}
            description={task?.description}
            closed={task?.closed}
          />
        ))}
      </ReactSortable>
    </List>
  );
});
