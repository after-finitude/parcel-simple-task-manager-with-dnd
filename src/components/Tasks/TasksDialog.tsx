import React from "react";
import { useSelector, shallowEqual } from "react-redux";

import { TasksCreateDialog } from "components/Tasks/TasksCreateDialog";
import { TasksUpdateDialog } from "components/Tasks/TasksUpdateDialog";

import { RootState } from "store/rootReducer";
import { DialogState, DialogTypes } from "store/dialog/types";

export const TasksDialog: React.FC = () => {
  const dialog = useSelector<RootState, DialogState>(
    state => state.dialog,
    shallowEqual,
  );

  if (dialog.type === DialogTypes.CREATE) {
    return <TasksCreateDialog status={dialog.status} />;
  }

  return (
    <TasksUpdateDialog
      status={dialog.status}
      currentTaskId={dialog.currentTaskId}
    />
  );
};
