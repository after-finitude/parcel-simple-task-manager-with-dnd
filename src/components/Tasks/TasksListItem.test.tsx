import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";

import { TasksListItem } from "components/Tasks/TasksListItem";

import { mockStore } from "utils/mocks/mockStore";

describe("TasksListItem", () => {
  it("should render correctly", () => {
    const component = mount(
      <Provider store={mockStore}>
        <TasksListItem id="test1" title="Test1" closed={false} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
