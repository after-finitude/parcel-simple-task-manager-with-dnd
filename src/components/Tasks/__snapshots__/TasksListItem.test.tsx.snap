// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksListItem should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo()
    closed={false}
    id="test1"
    title="Test1"
  >
    <styled.div
      closed={false}
    >
      <div
        className="sc-bdfBwQ GSvvC"
      >
        <WithStyles(ForwardRef(ListItem))>
          <ForwardRef(ListItem)
            classes={
              Object {
                "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                "button": "MuiListItem-button",
                "container": "MuiListItem-container",
                "dense": "MuiListItem-dense",
                "disabled": "Mui-disabled",
                "divider": "MuiListItem-divider",
                "focusVisible": "Mui-focusVisible",
                "gutters": "MuiListItem-gutters",
                "root": "MuiListItem-root",
                "secondaryAction": "MuiListItem-secondaryAction",
                "selected": "Mui-selected",
              }
            }
          >
            <li
              className="MuiListItem-container"
            >
              <div
                className="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
                disabled={false}
              >
                <WithStyles(ForwardRef(ListItemText))
                  key=".0"
                  primary="Test1"
                  secondary=""
                >
                  <ForwardRef(ListItemText)
                    classes={
                      Object {
                        "dense": "MuiListItemText-dense",
                        "inset": "MuiListItemText-inset",
                        "multiline": "MuiListItemText-multiline",
                        "primary": "MuiListItemText-primary",
                        "root": "MuiListItemText-root",
                        "secondary": "MuiListItemText-secondary",
                      }
                    }
                    primary="Test1"
                    secondary=""
                  >
                    <div
                      className="MuiListItemText-root MuiListItemText-multiline"
                    >
                      <WithStyles(ForwardRef(Typography))
                        className="MuiListItemText-primary"
                        component="span"
                        display="block"
                        variant="body1"
                      >
                        <ForwardRef(Typography)
                          className="MuiListItemText-primary"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          component="span"
                          display="block"
                          variant="body1"
                        >
                          <span
                            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                          >
                            Test1
                          </span>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                      <WithStyles(ForwardRef(Typography))
                        className="MuiListItemText-secondary"
                        color="textSecondary"
                        display="block"
                        variant="body2"
                      >
                        <ForwardRef(Typography)
                          className="MuiListItemText-secondary"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="textSecondary"
                          display="block"
                          variant="body2"
                        >
                          <p
                            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                          />
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </div>
                  </ForwardRef(ListItemText)>
                </WithStyles(ForwardRef(ListItemText))>
              </div>
              <WithStyles(ForwardRef(ListItemSecondaryAction))
                key=".1"
              >
                <ForwardRef(ListItemSecondaryAction)
                  classes={
                    Object {
                      "root": "MuiListItemSecondaryAction-root",
                    }
                  }
                >
                  <div
                    className="MuiListItemSecondaryAction-root"
                  >
                    <WithStyles(ForwardRef(IconButton))
                      aria-label="edit"
                      disabled={false}
                      edge="end"
                      onClick={[Function]}
                    >
                      <ForwardRef(IconButton)
                        aria-label="edit"
                        classes={
                          Object {
                            "colorInherit": "MuiIconButton-colorInherit",
                            "colorPrimary": "MuiIconButton-colorPrimary",
                            "colorSecondary": "MuiIconButton-colorSecondary",
                            "disabled": "Mui-disabled",
                            "edgeEnd": "MuiIconButton-edgeEnd",
                            "edgeStart": "MuiIconButton-edgeStart",
                            "label": "MuiIconButton-label",
                            "root": "MuiIconButton-root",
                            "sizeSmall": "MuiIconButton-sizeSmall",
                          }
                        }
                        disabled={false}
                        edge="end"
                        onClick={[Function]}
                      >
                        <WithStyles(ForwardRef(ButtonBase))
                          aria-label="edit"
                          centerRipple={true}
                          className="MuiIconButton-root MuiIconButton-edgeEnd"
                          disabled={false}
                          focusRipple={true}
                          onClick={[Function]}
                        >
                          <ForwardRef(ButtonBase)
                            aria-label="edit"
                            centerRipple={true}
                            className="MuiIconButton-root MuiIconButton-edgeEnd"
                            classes={
                              Object {
                                "disabled": "Mui-disabled",
                                "focusVisible": "Mui-focusVisible",
                                "root": "MuiButtonBase-root",
                              }
                            }
                            disabled={false}
                            focusRipple={true}
                            onClick={[Function]}
                          >
                            <button
                              aria-label="edit"
                              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label"
                              >
                                <ForwardRef>
                                  <WithStyles(ForwardRef(SvgIcon))>
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                              <WithStyles(memo)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(memo)>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                    <WithStyles(ForwardRef(IconButton))
                      aria-label="close"
                      edge="end"
                      onClick={[Function]}
                    >
                      <ForwardRef(IconButton)
                        aria-label="close"
                        classes={
                          Object {
                            "colorInherit": "MuiIconButton-colorInherit",
                            "colorPrimary": "MuiIconButton-colorPrimary",
                            "colorSecondary": "MuiIconButton-colorSecondary",
                            "disabled": "Mui-disabled",
                            "edgeEnd": "MuiIconButton-edgeEnd",
                            "edgeStart": "MuiIconButton-edgeStart",
                            "label": "MuiIconButton-label",
                            "root": "MuiIconButton-root",
                            "sizeSmall": "MuiIconButton-sizeSmall",
                          }
                        }
                        edge="end"
                        onClick={[Function]}
                      >
                        <WithStyles(ForwardRef(ButtonBase))
                          aria-label="close"
                          centerRipple={true}
                          className="MuiIconButton-root MuiIconButton-edgeEnd"
                          disabled={false}
                          focusRipple={true}
                          onClick={[Function]}
                        >
                          <ForwardRef(ButtonBase)
                            aria-label="close"
                            centerRipple={true}
                            className="MuiIconButton-root MuiIconButton-edgeEnd"
                            classes={
                              Object {
                                "disabled": "Mui-disabled",
                                "focusVisible": "Mui-focusVisible",
                                "root": "MuiButtonBase-root",
                              }
                            }
                            disabled={false}
                            focusRipple={true}
                            onClick={[Function]}
                          >
                            <button
                              aria-label="close"
                              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label"
                              >
                                <ForwardRef>
                                  <WithStyles(ForwardRef(SvgIcon))>
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                              <WithStyles(memo)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(memo)>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                  </div>
                </ForwardRef(ListItemSecondaryAction)>
              </WithStyles(ForwardRef(ListItemSecondaryAction))>
            </li>
          </ForwardRef(ListItem)>
        </WithStyles(ForwardRef(ListItem))>
      </div>
    </styled.div>
  </Memo()>
</Provider>
`;
